backend/forecastbox/api/graph_building.py

```diff
@@ -0,0 +1,145 @@
+from itertools import groupby
+from collections import defaultdict
+from forecastbox.api.types.graph_building import (
+    ActionKind,
+    ActionFactoryId,
+    ActionFactoryCatalog,
+    GraphValidationExpansion,
+    ActionFactory,
+    ActionConfigurationOption,
+    GraphBuilder,
+)
+from forecastbox.api.types import RawCascadeJob
+from cascade.low.core import JobInstance
+
+# NOTE this will not be hardcoded like this, but partially hardcoded in submodules and extended by plugins
+catalog = ActionFactoryCatalog(
+    factories={
+        "model_forecast": ActionFactory(
+            kind="source",
+            title="Compute Model Forecast",
+            description="Download initial conditions, run model forecast",
+            configuration_options={
+                "model": ActionConfigurationOption(title="Model Name", description="Locally available checkpoint to run", value_type="str"),
+                "date": ActionConfigurationOption(
+                    title="Initial Conditions DateTime", description="DateTime of the initial conditions", value_type="datetime"
+                ),
+                "lead_time": ActionConfigurationOption(title="Lead Time", description="Lead Time of the forecast", value_type="int"),
+                "ensemble_members": ActionConfigurationOption(
```

Comment:
todo: allow for various configuration prefills, including both user prefills and admin mandates. It should be presumably doubly keyed, as in, given this ActionFactory with this value for this field, these fields are prefilled like this and these fields are not editable